### ISSUE TITLE
showInMenu property of GridActionsCellItem does not work if action is wrapped 

### ISSUE THREAD

**Issue by Nic-S** [mui/mui-x#5950]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

If we wrap the GridActionsCellItem in a new component, the showInMenu property doesn't work.

### Expected behavior 🤔

showInMenu works even if the GridActionsCellItem is wrapped in another component

### Steps to reproduce 🕹

Link to live example: https://codesandbox.io/s/kind-stonebraker-k4nhoo?file=/demo.tsx

### Context 🔦

I tried wrapping the action since on click a confirmation dialog is opened and I wanted to isolate the behavior.

```JSX
export default function ResendInviteMenuItem(): JSX.Element {
  const [openDialog, setOpenDialog] = React.useState(false);
  const intl = useIntl();

  const handleDialogClose = () => {
    setOpenDialog(false);
  };
  const onResendInviteClick = () => {
    setOpenDialog(true);
  };

  const handleConfirm = () => {
    // api call
    return null;
  };

  return (
    <>
      <GridActionsCellItem
        data-testid="resend-invite-menu-item"
        onClick={onResendInviteClick}
        label={intl.formatMessage({ id: 'supplier.user.resend.invite' })}
        icon={<Redo />}
        showInMenu
      />
      {openDialog && (
        <ResendInviteConfirmationDialog
          isOpen={openDialog}
          onClose={handleDialogClose}
          onClick={handleConfirm}
        />
      )}
    </>
  );
}
```

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by DanailH** [comment]
Hi @Nic-S, thanks for raising this!
The problem is that our code looks for `showInMenu` prop on the top-level component that is passed to the `getActions` array. One solution to that problem is to add the `showInMenu` prop to your custom component and then pass it to the `GridActionsCellItem`. I've updated the Codesandbox so you can see what I have in mind -> https://codesandbox.io/s/amazing-euler-52o54g?file=/demo.tsx:2931-2950

**Comment by Nic-S** [comment]
Hi @DanailH , the example does not work correctly anymore, surely after updating the libraries (on package.json we take the latest). I can't use this solution if I risk that it breaks when updating the dataGrid.

**Comment by DanailH** [comment]
Hmm, interesting. It is broken indeed in the Codesandbox but it does seem to be working correctly here https://mui.com/x/react-data-grid/column-definition/#column-types
I assume you are using the latest v5, right?

**Comment by Nic-S** [comment]
@DanailH you are right, it is another regression. Yes i used the latest v5. I will open another issue.
